CANTroller2/src/TemperatureSensorManager.h

```diff
@@ -0,0 +1,142 @@
+#ifndef TEMPERATURESENSORS_H
+#define TEMPERATURESENSORS_H
+
+#include <vector>
+#include <map>
+#include <array>
+#include <algorithm>
+
+#include <OneWire.h>
+#include "temp.h"
+#include "globals.h"
+#include "utils.h"
+
+// Class to manage OneWire temperature sensors
+class TemperatureSensorManager {
+private:
+    using DeviceAddress = std::array<uint8_t, 8>;
+
+    static constexpr uint8_t ONEWIRE_PIN = onewire_pin; // todo read from config in the future
+    OneWire one_wire_bus;
+    DallasSensor tempsensebus;
+
+    int temp_detected_device_ct = 0;
+    std::vector<DeviceAddress> detected_addresses;
+    std::map<std::string, DeviceAddress> known_addresses = {
+        {"ENGINE", {0x28, 0x3c, 0xf3, 0xa7, 0x00, 0x00, 0x00, 0x00}},
```

Comment:
are we only gonna have these two temp sensors? or will we have the wheels as well?

Reply:
Once we know the addresses we can add them yeah. Until then, there's an "All_locations" that holds the places we know of, and if we detect an address that hasn't yet been matched to a location, we'll assign it to the next location in the list.